buildroot/share/PlatformIO/scripts/schema.py

```diff
@@ -133,11 +129,13 @@ class Parse:
         ERROR           = 9 # Syntax error
 
     # A JSON object to store the data
-    sch_out = { key:{} for key in filekey.values() }
+    sch_out = { key: {} for key in filekey.values() }
     # Regex for #define NAME [VALUE] [COMMENT] with sanitized line
     defgrep = re.compile(r'^(//)?\s*(#define)\s+([A-Za-z0-9_]+)\s*(.*?)\s*(//.+)?$')
     # Pattern to match a float value
-    flt = r'[-+]?\s*(\d+\.|\d*\.\d+)([eE][-+]?\d+)?[fF]?'
+    flt = r'[-+]?(?:\d+\.|\d*\.\d+)(?:[eE][-+]?\d+)?[fF]?'
```

Comment:
It is perfectly legal for a float value to have whitespace between the leading sign (`+` or `-`) and the first digit.